cle/backends/elf.py

```diff
@@ -211,6 +212,13 @@ def __init__(self, binary, **kwargs):
 
         self.__parsed_reloc_tables = set()
 
+        # The linked image base should be evaluated before registering any segment or section due to
+        # the fact that elffile, used by those methods, is working only with un-based virtual addresses, but Clemories
+        # themselves are organized as a tree where each node backer internally uses relative addressing
+        self.mapped_base = self.linked_base = min(
+            map(lambda x: x['p_vaddr'],
+                filter(lambda x: x.header.p_type == 'PT_LOAD', self.reader.iter_segments())) or [0]
+        )
```

Comment:
> Also take into account sections as well, if segments are not present.

I think there is a naming inconsistency between PE and ELF -- entities which are sections in PE are segments in case of ELF. There are also so called directories described in PE's optional header which seems equivalent to ELF sections. 

Maybe I am wrong, but loadable sections (in a case of ELF) should be always a part of loadable segments, so I don't understand how I can take sections into account and think the code is correct. 
If I am not then it is possible for valid ELF, that loadable sections exist without a loadable segment. Is it so? Because this is not in a case of valid PE.